mgmtfn/dockplugin/netDriver.go

```diff
@@ -523,3 +630,171 @@ func GetDockerNetworkName(nwID string) (string, string, string, error) {
+		return nil, err
+	}
+	return nw, nil
+}
+
+// createNetworkHelper creates the association between docker network and contiv network
+//  if tag is given map docker net to epg or network, else create a contiv network
+func createNetworkHelper(networkID string, tag string, IPv4Data, IPv6Data []driverapi.IPAMData) error {
+	var tenantName, networkName, serviceName string
+	var err error
+	if tag != "" {
+		// we need to map docker network to policy group or network using the tag
+		log.Infof("Received tag %s", tag)
+		var nw *mastercfg.CfgNetworkState
+		epg, err := FindGroupFromTag(tag)
+		if err != nil {
+			nw, err = FindNetworkFromTag(tag)
+			if err != nil {
+				return errors.New("failed to lookup tag")
+			}
+		}
+		if epg != nil {
+			tenantName = epg.TenantName
+			networkName = epg.NetworkName
+			serviceName = epg.GroupName
+		} else if nw != nil {
+			tenantName = nw.Tenant
+			networkName = nw.NetworkName
+			serviceName = ""
+		}
+	} else if len(IPv4Data) > 0 {
+		// if subnet is specified in docker command, we create a contiv network
+		subnetPool := ""
+		gateway := ""
+		if IPv4Data[0].Pool != nil {
+			subnetPool = IPv4Data[0].Pool.String()
+		}
+		if IPv4Data[0].Gateway != nil {
+			gateway = strings.Split(IPv4Data[0].Gateway.String(), "/")[0]
+		}
+		subnetv6 := ""
+		gatewayv6 := ""
+		if len(IPv6Data) > 0 {
+			if IPv6Data[0].Pool != nil {
+				subnetv6 = IPv6Data[0].Pool.String()
+			}
+			if IPv6Data[0].Gateway != nil {
+				gatewayv6 = strings.Split(IPv6Data[0].Gateway.String(), "/")[0]
+			}
+		}
+		// build key and URL
+		keyStr := "default" + ":" + networkID
+		url := "/api/v1/networks/" + keyStr + "/"
+
+		tenantName = "default"
+		networkName = networkID
+		serviceName = ""
+
+		req := client.Network{
+			TenantName:  tenantName,
+			NetworkName: networkName,
+			Subnet:      subnetPool,
+			Gateway:     gateway,
+			Ipv6Subnet:  subnetv6,
+			Ipv6Gateway: gatewayv6,
+			Encap:       "vlan",
+		}
+
+		var resp client.Network
+		err = cluster.MasterPostReq(url, &req, &resp)
+		if err != nil {
+			return errors.New("failed to create network")
+		}
+		log.Infof("Created contiv network %+v", req)
+	}
+	// Create docknet oper state to map the docker network to contiv network
+	// We do not create a network in docker as it is created explicitly by user
+	err = docknet.CreateDockNetState(tenantName, networkName, serviceName, networkID)
+	if err != nil {
+		log.Errorf("Error creating docknet state. ")
+		return errors.New("Error creating docknet state")
+	}
+	return nil
+}
+
+// deleteNetworkHelper removes the association between docker network and contiv network
+func deleteNetworkHelper(networkID string) error {
+	netID := networkID + ".default"
+	_, err := netdGetNetwork(netID)
+	if err == nil {
+		// if we find a contiv network with the ID hash, then it must be
+		// a docker created network (from the libnetwork create api).
+		// build key and URL
+		keyStr := "default" + ":" + networkID
```

Comment:
tenant name is also not known in the case of docker network created without tag, hence the default.

In this file netID is contiv ID/key for the network and networkID is the docker ID for the network. I used the same convention.